packages/block-editor/src/store/selectors.js

```diff
@@ -2167,7 +2163,7 @@ export const __experimentalGetAllowedBlocks = createSelector(
 			return;
 		}
 
-		return filter( getBlockTypes(), ( blockType ) =>
+		return getBlockTypes().filter( ( blockType ) =>
```

Comment:
`getBlockTypes()` will always be an array:

https://github.com/WordPress/gutenberg/blob/93c5efd7ebbebf0ae73247866e5f12589887ede0/packages/blocks/src/store/selectors.js#L71